optaplanner-website-root/content/blog/2022/09/26/DynamicScalingOfOptaPlannerWorkloadOnOpenShift.adoc

```diff
@@ -0,0 +1,93 @@
+= Run OptaPlanner workloads on OpenShift, part II.
+rsynek
+2022-09-26
+:page-interpolate: true
+:jbake-type: post
+:jbake-tags: cloud, openshift, kubernetes, scaling
+
+In the first https://www.optaplanner.org/blog/2022/06/09/RunOptaPlannerWorkloadOnOpenShift.html[article about running OptaPlanner workloads on OpenShift],
+I've shown how to use the OptaPlanner operator to solve multiple datasets of the same planning problem by starting several pods on OpenShift.
+
+That implementation kept a number of nodes always around, even if they were idle. But why pay the bills for those? Since the previous article, the OptaPlanner operator supports dynamic scaling driven by the number of input datasets.
+
+The https://github.com/kiegroup/optaplanner-quickstarts/tree/development/technology/kubernetes[Kubernetes demo],
+based on the https://www.optaplanner.org/learn/useCases/schoolTimetabling.html[school timetabling] use case,
+has been updated to show the dynamic scaling.
+
+== Architecture
+
+The architecture has evolved, replacing some existing components and adding a new one, too.
+
+image::demoArchitecture.svg[OpenShift demo architecture diagram]
+
+The OptaPlanner operator depends on two other operators:
+
+* https://artemiscloud.io/[ArtemisCloud] operator, which creates ActiveMQ Artemis broker and queues
+* https://keda.sh/[KEDA] for scaling the solver deployment, represented by the `school-timetabling` pods in the picture
```

Comment:
Id like to make this parallel with the previous bullet. Does this make sense?
The https://keda.sh/[KEDA] operator for scaling